.github/workflows/notebook_tests.yaml

```diff
@@ -2,42 +2,76 @@ name: Notebook Tests
 
 on:
   workflow_call:
-
+    inputs:
+      status-filter:
+          description: 'Only nb tests with the given status will be executed.'
+          required: true
+          type: string
 jobs:
+  fetch-notebook-tests:
+    runs-on: ubuntu-24.04
+    name: Fetching tests
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Python & Poetry Environment
+        uses: exasol/python-toolbox/.github/actions/python-environment@v1
+        with:
+          python-version: "3.10"
+
+      - id: set-normal-tests
+        run: |
+          TESTS=`poetry run -- nox -s get-notebook-tests -- --test-status ${{ inputs.status-filter }} --test-classification normal`
+          echo "normal-tests=$TESTS" >> "$GITHUB_OUTPUT"
+      - id: set-large-tests
+        run: |
+          TESTS=`poetry run -- nox -s get-notebook-tests -- --test-status ${{ inputs.status-filter }} --test-classification large`
+          echo "large-tests=$TESTS" >> "$GITHUB_OUTPUT"
+
+    outputs:
+      normal-tests: ${{ steps.set-normal-tests.outputs.normal-tests }}
+      large-tests: ${{ steps.set-large-tests.outputs.large-tests }}
+
+
   notebook_tests-normal-runner:
+    needs: [fetch-notebook-tests]
```

Comment:
Theoretically we should be able to extract the content of the job and the large runner workflow into their own file. However, I am not sure at which nesting level we are? Maybe also something for the next PR.